hazelcast/src/main/java/com/hazelcast/internal/server/Server.java

```diff
@@ -39,33 +40,54 @@
      * Returns a {@link ServerConnectionManager} that is able to handle
      * connections for the given qualifier.
      *
-     * It is important to realize that the returned manager could also
+     * It is important to realize that the return\ed manager could also
      * manage other connections; so if you ask for a member connection manager
      * it could be that you get a connection manager that also handles client
      * connections. So if you would ask for {@link ServerConnectionManager#getConnections()}
      * it could contain a mixture of client and member connections.
      *
      * @param qualifier the EndpointQualifier used to select the right connection manager.
      * @return the relevant {@link ServerConnectionManager} that processes connections
-     * for the given an {@link EndpointQualifier}
+     * for the given an {@link EndpointQualifier}. It for MEMBER there will always be an
+     * ConnectionManager, but for the other ones null could be returned.
      */
     ServerConnectionManager getConnectionManager(EndpointQualifier qualifier);
 
     /**
-     * Returns all connections that have been successfully established.
+     * Returns all connections.
      *
-     * @return active connections
+     * This can be a relatively expensive operations the returned collection might be created
+     * on every invocation. So if you are just interested in count, have a look at
+     * {@link #connectionCount(Predicate)} method.
+     *
+     * @return the connections.
```

Comment:
a `@Nonnull` annotation in this method could help clarify API. It seems there were some methods expecting it might return `null` but that was not the case.